Modules/_winapi.c

```diff
@@ -140,7 +153,9 @@ overlapped_dealloc(OverlappedObject *self)
     if (self->write_buffer.obj)
         PyBuffer_Release(&self->write_buffer);
     Py_CLEAR(self->read_buffer);
-    PyObject_Del(self);
+    PyTypeObject *tp = PY_TYPE(self);
```

Comment:
Your PR fails to build on Windows:

_winapi.obj : error LNK2001: unresolved external symbol PY_TYPE [D:\a\cpython\cpython\PCbuild\pythoncore.vcxproj]

I guess that it should be Py_TYPE().

Did you try to build your PR locally? Do you run Windows?

Reply:
@vstinner I have a bad habit of pushing tiny changes without trying them.  I'll try to break this habit to not waste your or the CI time.